akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -319,7 +319,35 @@ final class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends
    * '''Cancels when''' downstream cancels
    */
   def mapConcat[T](f: function.Function[Out, java.lang.Iterable[T]]): javadsl.Flow[In, T, Mat] =
-    new Flow(delegate.mapConcat { elem ⇒ Util.immutableSeq(f(elem)) })
+    new Flow(delegate.statefulMapConcat { () ⇒ elem ⇒ Util.immutableSeq(f(elem)) })
+
+  /**
+    * Transform each input element into an `Iterable` of output elements that is
+    * then flattened into the output stream. Transformation is stateful
+    * meaning it's taking into account previous messages (hysteresis).
+    *
+    * Make sure that the `Iterable` is immutable or at least not modified after
+    * being used as an output sequence. Otherwise the stream may fail with
+    * `ConcurrentModificationException` or other more subtle errors may occur.
+    *
+    * The returned `Iterable` MUST NOT contain `null` values,
+    * as they are illegal as stream elements - according to the Reactive Streams specification.
+    *
+    * '''Emits when''' the mapping function returns an element or there are still remaining elements
+    * from the previously calculated collection
+    *
+    * '''Backpressures when''' downstream backpressures or there are still remaining elements from the
+    * previously calculated collection
+    *
+    * '''Completes when''' upstream completes and all remaining elements has been emitted
+    *
+    * '''Cancels when''' downstream cancels
+    */
+  def statefulMapConcat[T](f: function.Creator[function.Function[Out, java.lang.Iterable[T]]]):javadsl.Flow[In, T, Mat] =
+    new Flow(delegate.statefulMapConcat{ () ⇒ {
```

Comment:
Not sure I understand. I created additional function here for converting java Iterator to scala one. But perhaps you are talking about mapConcat?